components/script/dom/htmllinkelement.rs

```diff
@@ -79,11 +79,16 @@ impl HTMLLinkElement {
     }
 
     pub fn set_stylesheet(&self, s: Arc<Stylesheet>) {
-        assert!(self.stylesheet.borrow().is_none());
-        *self.stylesheet.borrow_mut() = Some(s);
+        let mut sheet = self.stylesheet.borrow_mut();
+        let must_invalidate = sheet.is_some();
```

Comment:
Note that we still need to support setting a stylesheet when it is loaded and the href changes, or at least (probably preferably?) add support for unsetting it and `invalidate_stylesheets` after the href changes.

So yeah, let's keep the assertion there, let's add the load counter, and also let's move this `must_invalidate` logic to `handle_stylesheet_url`, and unset the stylesheet there.